cli/src/program.rs

```diff
@@ -2636,10 +2641,11 @@ fn send_deploy_messages(
                     },
                 )
                 .map_err(|e| format!("Deploying program failed: {e}"))?;
-        }
-    }
+            Result::<_, Box<dyn std::error::Error>>::Ok(final_tx_sig)
+        })
+        .transpose()?;
 
-    Ok(())
+    Ok(final_transaction)
```

Comment:
What about making the final line `Ok(None)` and just doing this in the `if let` blocks:

```rust
return rpc_client
    .send_and_confirm_transaction_with_spinner_and_config(
        &final_tx,
        config.commitment,
        RpcSendTransactionConfig {
            preflight_commitment: Some(config.commitment.commitment),
            ..RpcSendTransactionConfig::default()
        },
    )
    .map(|s| Ok(Some(s)))
    .map_err(|e| format!("Deploying program failed: {e}"))?;
```